test/invites.js

```diff
@@ -187,3 +189,94 @@ describe('Member style channel init', () => {
+	let users = [
+		'thierry-' + uuidv4(),
+		'tommaso-' + uuidv4(),
+		'josh-' + uuidv4(),
+		'scott-' + uuidv4(),
+	];
+	let channelID = uuidv4();
+	let thierryClient;
+	let tommasoClient;
+	let joshClient;
+	let scottClient;
+
+	before(async () => {
+		await createUsers(users);
+		thierryClient = await getTestClientForUser(users[0]);
+		tommasoClient = await getTestClientForUser(users[1]);
+		joshClient = await getTestClientForUser(users[2]);
+		scottClient = await getTestClientForUser(users[3]);
+	});
+	it('Thierry creates a channel and invite Tommaso, Josh and Scott', async function() {
+		const c = thierryClient.channel('messaging', channelID, {
+			name: 'Founder Chat',
+			image: 'http://bit.ly/2O35mws',
+			members: users,
+			invites: [users[1], users[2], users[3]],
+		});
+		const state = await c.create();
+		expect(state.channel.id).to.be.equal(channelID);
+	});
+	it('Tommaso should have pending invites', async function() {
+		let channels = await tommasoClient.queryChannels({$invited: 'pending'});
+		expect(channels.length).to.be.equal(1);
+		expect(channels[0].id).to.be.equal(channelID);
+	});
+	it('Josh should have pending invites', async function() {
+		let channels = await joshClient.queryChannels({$invited: 'pending'});
+		expect(channels.length).to.be.equal(1);
+		expect(channels[0].id).to.be.equal(channelID);
+	});
+	it('Scott should have pending invites', async function() {
+		let channels = await scottClient.queryChannels({$invited: 'pending'});
+		expect(channels.length).to.be.equal(1);
+		expect(channels[0].id).to.be.equal(channelID);
+	});
+	it('Tommaso accept the invite and pending invites go to zero', async function() {
+		let channels = await tommasoClient.queryChannels({$invited: 'pending'});
+		await channels[0].acceptInvite();
+
+		channels = await tommasoClient.queryChannels({$invited: 'pending'});
+		expect(channels.length).to.be.equal(0);
+	});
+	it('Tommaso queries for accepted invites it should return one result', async function () {
+		let channels = await tommasoClient.queryChannels({$invited: 'accepted'});
+		expect(channels.length).to.be.equal(1);
+		expect(channels[0].id).to.be.equal(channelID);
+	});
+	it('Tommaso queries for pending invites it should return one result', async function () {
+		let channels = await tommasoClient.queryChannels({$invited: 'pending'});
+		expect(channels.length).to.be.equal(0);
+	});
+	it('Josh Reject the invite. the channel state is still available but watch:true and presence:true is a noop', async function() {
+		let channels = await joshClient.queryChannels({$invited: 'pending'});
+		await channels[0].rejectInvite();
+		await channels[0].stopWatching();
+
+		channels = await joshClient.queryChannels({$invited: 'pending'});
+		expect(channels.length).to.be.equal(0);
+
+		let rejectedChannel = joshClient.channel('messaging', channelID);
+		let resp = await rejectedChannel.watch({ presence: true });
+		let channelEventsReceived = 0;
+		rejectedChannel.on(function(e) {
```

Comment:
better way to wait:

```js
let doneCallback;
const messageHandled = new Promise(resolve => { doneCallback = resolve; });

rejectedChannel.on(function(e) {
  channelEventsReceived++;
  doneCallback();
});

await channel.sendMessage({ text: 'hi' });
await messageHandled;
```

Reply:
here we want to make sure that we don't receive an event. this will block forever right?

Reply:
nope there is a test timeout